proposals/0013-expanded-comparison-sampling.md

```diff
@@ -0,0 +1,432 @@
+    i32,                  ; opcode
+    %dx.types.Handle,     ; texture handle
+    %dx.types.Handle,     ; sampler handle
+    float,                ; coordinate c0
+    float,                ; coordinate c1
+    float,                ; coordinate c2
+    float,                ; coordinate c3
+    i32,                  ; offset o0
+    i32,                  ; offset o1
+    i32,                  ; offset o2
+    float,                ; compare value
+    float,                ; ddx0
+    float,                ; ddx1
+    float,                ; ddx2
+    float,                ; ddy0
+    float,                ; ddy1
+    float,                ; ddy2
+    float)                ; clamp
+```
+
+Valid resource types for the texture handle parameter and the corresponding
+ active coordinates and offsets are the same as for the `sampleGrad` operation
+ except that `TextureCubeArray` resources are invalid.
+Valid active ddx and ddy are the same as offsets.
+The sampler handle parameter must be of type `SamplerComparisonState`.
+
+```llvm
+declare %dx.types.ResRet.f32 @dx.op.sampleCmpBias.f32(
+    i32,                  ; opcode
+    %dx.types.Handle,     ; texture handle
+    %dx.types.Handle,     ; sampler handle
+    float,                ; coordinate c0
+    float,                ; coordinate c1
+    float,                ; coordinate c2
+    float,                ; coordinate c3
+    i32,                  ; offset o0
+    i32,                  ; offset o1
+    i32,                  ; offset o2
+    float,                ; compare value
+    float,                ; bias: in [-16.f,15.99f]
+    float)                ; clamp
+```
+
+Valid resource types for the texture handle parameter and the corresponding
+ active coordinates and offsets are the same as for the `sampleBias` operation
+ except that `TextureCubeArray` resources are invalid for `sampleCmpGrad`.
+The sampler handle parameter must be of type `SamplerComparisonState`.
+
+The existing `dx.op.calculateLOD` intrinsic is reused for the new
+ `CalculateLevelOfDetail` overloads taking `SamplerComparisonState` parameters.
+
+### SPIR-V Additions
+
+To implement the new `SampleCmpGrad` and `SampleCmpBias` texture methods,
+ the existing `OpImage*SampleDrefImplicitLod` operations can be used with the
+ optional image operands set to the value for `Grad`(0x4) or `Bias`(0x1) set
+ to the values provided by the user.
+
+To implement the new `CalculateLevelOfDetail[Unclamped]` overloads,
+ the existing `OpImageQueryLod` operation may be used with comparison state images.
+
+### Diagnostic Changes
+
+#### Removed errors
+
+* Using `CalculateLevelOfDetail[Unclamped]` with a `SamplerComparisonState`
+  sampler will no longer produce an error when compiling with appropriate shader
+  model versions.
+* Invoking new methods `SampleCmpBias` and `SampleCmpGrad` with valid texture
+  objects and parameters will not produce an unrecognized function error.
```

Comment:
I'm not sure how useful/necessary it is to list newly introduced intrinsic functions as if they are removed errors in this way.

Reply:
Removed